### PR TITLE
Update to work with influxdb 0.9

### PR DESCRIPTION
**Pull request by contentfree**

### CODE REVIEW

**Comment by suyograo** [comment]
@contentfree thanks! Can you please a) rebase? b) sign the CLA https://github.com/elasticsearch/logstash/blob/master/CONTRIBUTING.md#contribution-steps

**Comment by contentfree** [comment]
Hmm... I'd signed the CLA. Looking at the commit log above, I think that it might be another contributor to my fork that hasn't signed. Will rebase…

**Comment by contentfree** [comment]
I still need to run the tests to verify that they still work

**Comment by masonbryant** [comment]
Impatiently awaiting this fix! :)

It looks like this just needs a review, is this likely to be approved soon?

**Comment by dalegaspi** [comment]
oh pls pls i need dis!  thanks!

**Comment by noseka1** [comment]
+1

**Comment by aseppala** [comment]
+1

However I have a question how to configure this plugin to handle data that already comes in InfluxDB line format?

e.g.

```
system_load1,host=debian value=0 1449261770042724367
system_load5,host=debian value=0.01 1449261770042741446
system_load15,host=debian value=0.05 1449261770042747442
```

**Comment by timgriffiths** [comment]
I been testing out this branch, I think need to modify this a little bit to handle strings better or needs to stick them in double quotes so it can handle spaces in the string. https://influxdb.com/docs/v0.9/write_protocols/line.html

**Comment by gnewman7** [comment]
@timgriffiths  To handle a field called task_name that contains spaces I added the following filter to my conf file

mutate { gsub => ["task_name"," ","\ "]}

Here is the logstash reference to mutate.
https://www.elastic.co/guide/en/logstash/current/plugins-filters-mutate.html#plugins-filters-mutate-gsub

**Comment by timgriffiths** [comment]
Yep that's how i have worked around this for now, but i think the plugin should just handle this it's part of the protocol

**Comment by jordansissel** [comment]
jenkins, test this

File: lib/logstash/outputs/influxdb.rb [hunk 1]
```diff
@@ -20,7 +20,10 @@ class LogStash::Outputs::InfluxDB < LogStash::Outputs::Base
   config_name "influxdb"
 
   # The database to write
-  config :db, :validate => :string, :default => "stats"
+  config :db, :validate => :string, :default => "statistics"
```
Comment (jordansissel) [inline lib/logstash/outputs/influxdb.rb]: (caveat, I have no experience with influxdb)

This changes the default db name which I would guess causes anyone who upgrades to this plugin to leave their old dataset behind - is this right? This seems a change that breaks backwards compatibility.

File: lib/logstash/outputs/influxdb.rb [hunk 2]
```diff
@@ -29,18 +32,18 @@ class LogStash::Outputs::InfluxDB < LogStash::Outputs::Base
...
+  config :user, :validate => :string, :default => nil
 
   # The password for the user who access to the named database
-  config :password, :validate => :password, :default => nil, :required => true
+  config :password, :validate => :password, :default => nil
 
-  # Series name - supports sprintf formatting
-  config :series, :validate => :string, :default => "logstash"
+  # Measurement name - supports sprintf formatting
+  config :measurement, :validate => :string, :default => "logstash"
```
Comment (jordansissel) [inline lib/logstash/outputs/influxdb.rb]: breaking change, renaming 'series' to 'measurement'

We'll want to bump the major version of this plugin as a result.

File: lib/logstash/outputs/influxdb.rb [hunk 3]
```diff
@@ -57,7 +60,7 @@ class LogStash::Outputs::InfluxDB < LogStash::Outputs::Base
   # Set the level of precision of `time`
   #
   # only useful when overriding the time value
-  config :time_precision, :validate => ["m", "s", "u"], :default => "s"
+  config :time_precision, :validate => ["n", "u", "ms", "s", "m", "h"], :default => "ms"
```
Comment (jordansissel) [inline lib/logstash/outputs/influxdb.rb]: breaking change by changing the default value. Tell me a story about why change the default value?

File: lib/logstash/outputs/influxdb.rb [hunk 4]
```diff
@@ -84,15 +105,17 @@ class LogStash::Outputs::InfluxDB < LogStash::Outputs::Base
...
+    require 'manticore'
     require 'cgi'
-    @agent = FTW::Agent.new
+    
+    @client = Manticore::Client.new
     @queue = []
 
-    @query_params = "u=#{@user}&p=#{@password.value}&time_precision=#{@time_precision}"
-    @base_url = "http://#{@host}:#{@port}/db/#{@db}/series"
+    @query_params = "db=#{@db}&rp=#{@retention_policy}&precision=#{@time_precision}&u=#{@user}&p=#{@password.value}"
```
Comment (jordansissel) [inline lib/logstash/outputs/influxdb.rb]: both user and password are made optional in this PR, so `@user` and `@password.value` can be nil here. Is that ok?

**Comment by jordansissel** [comment]
Code overall LGTM. I left a few inline comments about backwards compatiibily and versioning.

Looks like the test suite failed - [check out the failure](http://build-eu-00.elastic.co/job/logstash-plugin-output-influxdb_PR/1/console) under 'Failures:' - one test is failing:

```
es:

  1) LogStash::Outputs::InfluxDB complete pipeline run with 2 events should receive 2 events, flush and call post with 2 items json array
     Failure/Error: subject.close
       LogStash::Outputs::InfluxDB: {"host"=>"localhost", "user"=>"someuser", "password"=><password>, "allow_time_override"=>true, "data_points"=>{"foo"=>"%{foo}", "bar"=>"%{bar}", "time"=>"%{time}"}, "codec"=><LogStash::Codecs::Plain charset=>"UTF-8">, "workers"=>1, "db"=>"statistics", "retention_policy"=>"default", "port"=>8086, "measurement"=>"logstash", "time_precision"=>"ms", "coerce_values"=>{}, "use_event_fields_for_data_points"=>false, "exclude_fields"=>["@timestamp", "@version", "sequence", "message", "type"], "send_as_tags"=>["host"], "flush_size"=>100, "idle_flush_time"=>1} received :post with unexpected arguments
         expected: ("[{\"name\":\"logstash\",\"columns\":[\"foo\",\"bar\",\"time\"],\"points\":[[\"1\",\"2\",\"3\"],[\"1\",\"2\",\"3\"]]}]")
              got: ("logstash foo=\"1\",bar=\"2\" 3\nlogstash foo=\"1\",bar=\"2\" 3")
        Please stub a default value first if message might be received with other args as well. 
     # ./lib/logstash/outputs/influxdb.rb:171:in `flush'
     # ./lib/logstash/outputs/influxdb.rb:212:in `close'
     # ./spec/outputs/influxdb_spec.rb:36:in `(root)'

Finished in 0.827 seconds (files took 2.37 seconds to load)
5 examples, 1 failure
```

**Comment by gnewman7** [comment]
@jordansissel 
The test failure that expected:
 ("[{\"name\":\"logstash\",\"columns\":[\"foo\",\"bar\",\"time\"],\"points\":[[\"1\",\"2\",\"3\"],[\"1\",\"2\",\"3\"]]}]")  <---- This is looks like the old influxdb 0.8.x write protocol

For influxdb 0.9.x the write line protocol has changed what you got 
got: ("logstash foo=\"1\",bar=\"2\" 3\nlogstash foo=\"1\",bar=\"2\" 3")
Here is the documentation links
https://docs.influxdata.com/influxdb/v0.9/write_protocols/write_syntax/
https://docs.influxdata.com/influxdb/v0.9/write_protocols/json/

Hope that helps

**Comment by timgriffiths** [comment]
@contentfree the first until test is broken as it's still expecting the result to be json but you have converted it all across to the new line protocol

**Comment by Kasen** [comment]
@jordansissel what's status of this PR?

**Comment by CamJN** [comment]
👍🏻

**Comment by dalegaspi** [comment]
merge! merge! merge!

**Comment by suyograo** [comment]
@timgriffiths can you sign the CLA since you have a commit for he broken unit tests? This is step2 of https://github.com/elasticsearch/logstash/blob/master/CONTRIBUTING.md#contribution-steps.

@contentfree FYI ^^ 

We will not be able to merge until the CLA has been signed, or the commit fc3549d has been replaced by someone who has signed the CLA

**Comment by timgriffiths** [comment]
@suyograo So i have signed the CLA, I have also created #30 which covers all the good work @contentfree present in this pull, but I have fixed the unit test and handle spaces and comma's and equal signs according to the influx line protocol all unit tested out as well.

**Comment by IngaFeick** [comment]
+1

**Comment by alfredocambera** [comment]
Hope this gets merged soon. Using tag keys would be very useful.

**Comment by zealpath** [comment]
+1

**Comment by xo4n** [comment]
+1

**Comment by Kasen** [comment]
:100:

**Comment by Kasen** [comment]
Guys, someone must do something with this PR.

**Comment by alfredocambera** [comment]
influxdb 0.10 was released a few days ago...

**Comment by suyograo** [comment]
@contentfree and @timgriffiths finally merged https://github.com/logstash-plugins/logstash-output-influxdb/pull/30. Thanks for your patience. 

@contentfree and @timgriffiths would you be interested in helping maintain this plugin?

**Pull request merged**